src/Shuttle.h

```diff
@@ -56,34 +107,51 @@ class CommandParameters;
-   bool *pOptionalFlag;
-   CommandParameters * mpEap;
-   ShuttleParams() { mpEap = NULL; pOptionalFlag = NULL; }
-   virtual ~ShuttleParams() {}
+   std::conditional_t<Const, const bool, bool> *pOptionalFlag{};
+   CommandParameters * mpEap{};
+
+   SettingsVisitorBase() {}
+   virtual ~SettingsVisitorBase();
+
    bool ShouldSet();
-   virtual ShuttleParams & Optional( bool & WXUNUSED(var) ){ pOptionalFlag = NULL;return *this;};
-   virtual ShuttleParams & OptionalY( bool & var ){ return Optional( var );};
-   virtual ShuttleParams & OptionalN( bool & var ){ return Optional( var );};
-   virtual void Define( bool & var,     const wxChar * key, const bool vdefault, const bool vmin=false, const bool vmax=false, const bool vscl=false );
-   virtual void Define( size_t & var,   const wxChar * key, const int vdefault, const int vmin=0, const int vmax=100000, const int vscl=1 );
-   virtual void Define( int & var,      const wxChar * key, const int vdefault, const int vmin=0, const int vmax=100000, const int vscl=1 );
-   virtual void Define( float & var,    const wxChar * key, const float vdefault, const float vmin, const float vmax, const float vscl=1.0f );
-   virtual void Define( double & var,   const wxChar * key, const float vdefault, const float vmin, const float vmax, const float vscl=1.0f );
-   virtual void Define( double & var,   const wxChar * key, const double vdefault, const double vmin, const double vmax, const double vscl=1.0f );
-   virtual void Define( wxString &var, const wxChar * key, const wxString vdefault, const wxString vmin = {}, const wxString vmax = {}, const wxString vscl = {} );
-   virtual void DefineEnum( int &var, const wxChar * key, const int vdefault,
+   virtual SettingsVisitorBase &Optional( Ref<bool> var );
+   virtual SettingsVisitorBase &OptionalY( Ref<bool> var );
+   virtual SettingsVisitorBase &OptionalN( Ref<bool> var );
+   virtual void Define( Arg<bool> var, const wxChar * key, bool vdefault,
+      bool vmin = false, bool vmax = false, bool vscl = false );
+   virtual void Define( Arg<size_t> var, const wxChar * key, int vdefault,
+      int vmin = 0, int vmax = 100000, int vscl = 1 );
+   virtual void Define( Arg<int> var, const wxChar * key, int vdefault,
+      int vmin = 0, int vmax = 100000, int vscl = 1 );
+   virtual void Define( Arg<float> var, const wxChar * key, float vdefault,
+      float vmin, float vmax, float vscl = 1.0f );
+   virtual void Define( Arg<double> var, const wxChar * key, float vdefault,
+      float vmin, float vmax, float vscl = 1.0f );
+   virtual void Define( Arg<double> var, const wxChar * key, double vdefault,
+      double vmin, double vmax, double vscl = 1.0f );
+   virtual void Define( Ref<wxString> var, const wxChar * key,
+      wxString vdefault,
+      wxString vmin = {}, wxString vmax = {},
+      wxString vscl = {} );
+   virtual void DefineEnum( Arg<int> var, const wxChar * key, int vdefault,
       const EnumValueSymbol strings[], size_t nStrings );
 };
 
```

Comment:
On these two I get this warnings on msvc2019:

`warning C4910: 'SettingsVisitorBase<false>': '__declspec(dllexport)' and 'extern' are incompatible on an explicit instantiation`

Not sure how serious this warning is though: I can not get to the link stage because I get a compile error (which I noted elsewhere)